src/core/net/dnssd_server.hpp

```diff
@@ -107,27 +119,36 @@ class Server : public InstanceLocator, private NonCopyable
      * This method notifies a discovered host.
      *
      * @param[in] aHostFullName     The null-terminated full host name.
-     * @param[in] aHostInfo         A pointer to the discovered host information.
+     * @param[in] aHostInfo         A reference to the discovered host information.
      *
      */
     void HandleDiscoveredHost(const char *aHostFullName, const otDnssdHostInfo &aHostInfo);
 
-private:
-    enum
-    {
-        kPort                 = OPENTHREAD_CONFIG_DNSSD_SERVER_PORT,
-        kProtocolLabelLength  = 4,
-        kMaxConcurrentQueries = 32,
-    };
+    /**
+     * This function aquires the next query in the server.
+     *
+     * @param[in] aQuery            The query pointer. Pass nullptr to get the first query.
+     *
+     * @returns  A pointer to the query or nullptr if no more queries.
+     *
+     */
+    const otDnssdQuery *GetNextQuery(const otDnssdQuery *aQuery) const;
+
+    /**
+     * This function aquires the dns-sd query type and name for a specific query.
+     *
+     * @param[in]   aQuery            The query pointer.
+     * @param[out]  aNameOutput       The name output buffer.
+     *
+     * @returns The dns-sd query type.
+     *
+     */
+    static DnsQueryType GetQueryTypeAndName(const otDnssdQuery *aQuery, char (&aName)[Name::kMaxNameSize]);
```

Comment:
```suggestion
    static DnsQueryType GetQueryTypeAndName(const otDnssdQuery &aQuery, char (&aName)[Name::kMaxNameSize]);
```
nit: use reference

Reply:
Since `otDnssdQuery` is in fact `void`, it's not possible to have a void reference.